src/App.js

```diff
@@ -1,25 +1,74 @@
-import logo from './logo.svg';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+  Link,
+} from 'react-router-dom';
 import './App.css';
+import { useState } from 'react';
+import HomePage from './HomePage';
+import ListPage from './ListPage';
+import CreatePage from './CreatePage';
+import UpdatePage from './UpdatePage';
+import { logout } from './services/Fetch-utils';
+import { client } from './services/client';
+
+export default function App() {
+  const [user, setUser] = useState(client.auth.user());
+  
+  async function handleLogoutClick() {
+    await logout();
+    setUser('');
+  }
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+    <Router>
+      <div className='links'>
+        <nav>
+          <ul>
+            <li>
+              <Link to="/">Sign In</Link>
+            </li>
+            <li>
+              <Link to="/create">Create New Movie</Link>
+            </li>
+            <li>
+              <Link to="/movies/1">Update a Movie</Link>
+            </li>
+            <li>
+              <Link to="/movies">List of Your Movies</Link>
+            </li>
+            <li>
+              {user && 
+            <button onClick={handleLogoutClick}>Logout</button>}
+            </li>
+          </ul>
+        </nav>
 
-export default App;
+        <Switch>
+          <Route exact path="/">
+            {
+              !user
+                ? <HomePage setUser={setUser} />
+                : <Redirect to="/movies" />
+            }
+          </Route>
+          <Route exact path="/movies/:id">
+            <UpdatePage />
+          </Route>
+          <Route exact path="/movies">
+            {
+              user 
+                ? <ListPage /> 
+                : <Redirect to="/" />
+            }
+          </Route>
+          <Route exact path="/create">
```

Comment:
you can protect your create, and update routes similarly to how the List Page is being protected with this ternary on line 62